app/main/views/applications.py

```diff
@@ -169,7 +169,10 @@ def delete_application(application_id):
 def applications_list_response(with_task_status=False):
     page = get_valid_page_or_1()
 
-    applications = Application.query.filter(Application.status != 'deleted')
+    if request.args.get('status'):
```

Comment:
was thinking about this
wouldnt it be better to include filter in path eg:
/applications/<path:status>